File: ruby/resistor-color-duo/resistor_color_duo.rb
module ResistorColorDuo
  COLORS = {
    "black" => 0,
    "brown" => 1,
    "red" => 2,
    "orange" => 3,
    "yellow" => 4,
    "green" => 5,
    "blue" => 6,
    "violet" => 7,
    "grey" => 8,
    "white" => 9
  }

  def self.value(arr)
    arr
      .reverse_each
      .with_index
      .sum { |n, i| COLORS[n] * 10 ** i }
  end
end

File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Map;

class ResistorColorDuo {
    private static Map<String, Integer> colorCodes = Map.of(
        "black", 0,
        "brown", 1,
        "red", 2,
        "orange", 3,
        "yellow", 4,
        "green", 5,
        "blue", 6,
        "violet", 7,
        "grey", 8,
        "white", 9
    );

    int value(String[] colors) {
        return colorCodes.get(colors[1]) + colorCodes.get(colors[0]) * 10;
    }
}
